include/spdlog/details/thread_pool-inl.h

```diff
@@ -101,26 +106,34 @@ bool SPDLOG_INLINE thread_pool::process_next_msg_() {
     async_msg incoming_async_msg;
     q_.dequeue(incoming_async_msg);
 
+    bool active = true;
     switch (incoming_async_msg.msg_type) {
         case async_msg_type::log: {
             incoming_async_msg.worker_ptr->backend_sink_it_(incoming_async_msg);
-            return true;
+            active = true;
+            break;
         }
         case async_msg_type::flush: {
             incoming_async_msg.worker_ptr->backend_flush_();
-            return true;
+            active = true;
+            break;
         }
 
         case async_msg_type::terminate: {
-            return false;
+            active = false;
+            break;
         }
 
         default: {
             assert(false);
         }
     }
 
-    return true;
+    if (incoming_async_msg.promise) {
```

Comment:
No need for all those changes. since promise only used in flush, it can be dealt directly in the switch case and not involve the rest of the code. Also for clarity I would call the struct member it async_msg.flush _promise since this is its prupose.

Reply:
Done.